test/app-luatest/gh_7808_rel_paths_in_diag_and_debug_info_test.lua

```diff
@@ -0,0 +1,21 @@
+local netbox = require('net.box')
+local t = require('luatest')
+
+local g = t.group()
+
+g.test_rel_paths_in_diag = function(_)
+    local _, err = pcall(function() netbox.self:call('') end)
+    t.assert_not(err.trace[1].file:startswith('/'))
+end
+
+g.test_rel_paths_in_debug_info = function(_)
+    t.skip_if(bit.rshift(bit.band(os.execute('readelf --version'), 0xFF00), 8) == 127)
```

Comment:
This check looks hackish. `readelf` should be available on any Linux distribution that has a compiler installed. Let's change it to `require('jit').os ~= 'Linux'`?

Reply:
> This check looks hackish.

Sounds subjective. I use standard utilities to check availability of a program so that we can test a wider range of platforms. Moreover, relying on OS here is kind of strange.

Reply:
The check looks hackish, because of the bit manipulations with the return code. These manipulations are, in fact, OS-dependent according to the Lua manual:

[![image](https://user-images.githubusercontent.com/1114511/198966518-56230789-3b42-42f4-b5db-d9b68317d9ea.png)](https://www.lua.org/manual/5.1/manual.html).

Reply:
Discussed f2f: decided to skip the test on `macOS` and with DEV_BUILD option enabled.